web/viewer.css

```diff
@@ -1134,8 +1138,20 @@ a:is(.toolbarButton, .secondaryToolbarButton)[href="#"] {
   cursor: default;
 }
 
-#numPages.toolbarLabel {
-  padding-inline-start: 3px;
+#numPages {
+  &.toolbarLabel {
+    padding-inline-start: 3px;
+  }
+
+  &::before {
+    inset-inline-start: -54px;
```

Comment:
Couldn't we remove this `54` in putting the input and the span in a div (or span) and then in adding the icon in ::before with a position absolute and a margin... something like that ?

Reply:
I did think about that, but I'm not sure if it's actually an overall simplification though.

Reply:
I think it makes sense to have a container for these two elements because at the end they're finally one component (e.g. pageNumberDisplayer) and removing the `54`really  sounds like a simplification.

Reply:
I've just tried this, and it seems to lead to a number of problems with the icon not being visible (it ends up *behind* the input-elements) and it generally seems way more complicated like that.

It's certainly possible that I've done things incorrectly, but I cannot *easily* make your suggestion work unfortunately!

Reply:
I just tested this code:
```diff
diff --git a/web/viewer.css b/web/viewer.css
index f323952b6..066937d24 100644
--- a/web/viewer.css
+++ b/web/viewer.css
@@ -1138,20 +1138,22 @@ a:is(.toolbarButton, .secondaryToolbarButton)[href="#"] {
   cursor: default;
 }
 
+#pageNumberContainer::before {
+    content: "";
+    position: absolute;
+    width: 16px;
+    height: 16px;
+    top: calc(50% - 8px);
+    inset-inline-start: 2px;
+    background-color: var(--toolbar-icon-bg-color);
+    mask-size: cover;
+    mask-image: var(--loading-icon);
+}
+
 #numPages {
   &.toolbarLabel {
     padding-inline-start: 3px;
   }
-
-  &::before {
-    inset-inline-start: -54px;
-
-    transition-property: visibility;
-    transition-delay: var(--loading-icon-delay);
-  }
-  #pageNumber.loading + &::before {
-    visibility: visible;
-  }
 }
 
 #thumbnailView,
diff --git a/web/viewer.html b/web/viewer.html
index 46a767396..530f0efa8 100644
--- a/web/viewer.html
+++ b/web/viewer.html
@@ -325,8 +325,10 @@ See https://github.com/adobe-type-tools/cmap-resources
                     <span data-l10n-id="pdfjs-next-button-label">Next</span>
                   </button>
                 </div>
-                <input type="number" id="pageNumber" class="toolbarField" title="Page" value="1" min="1" tabindex="15" data-l10n-id="pdfjs-page-input" autocomplete="off">
-                <span id="numPages" class="toolbarLabel"></span>
+                <span id="pageNumberContainer">
+                  <input type="number" id="pageNumber" class="toolbarField" title="Page" value="1" min="1" tabindex="15" data-l10n-id="pdfjs-page-input" autocomplete="off">
+                  <span id="numPages" class="toolbarLabel"></span>
+                </span>
               </div>
               <div id="toolbarViewerRight">
                 <button id="print" class="toolbarButton hiddenMediumView" title="Print" tabindex="32" data-l10n-id="pdfjs-print-button">
```
and it wfm.

Reply:
That might work in "isolation", however there unfortunately seem to be two problems with it:
 - The same approach doesn't appear to work in the findbar (using the existing `findbarInputContainer`).
 - It affects the spacing between the `pageNumber` and `numPages` elements.

Hence it seems to me that this would require a bunch of additional changes/tweaks, and it thus feels like a can of worms that I'd rather not open here.

Reply:
What about:
```diff
diff --git a/web/viewer.css b/web/viewer.css
index f323952b6..d72b868ca 100644
--- a/web/viewer.css
+++ b/web/viewer.css
@@ -1138,19 +1138,30 @@ a:is(.toolbarButton, .secondaryToolbarButton)[href="#"] {
   cursor: default;
 }
 
-#numPages {
-  &.toolbarLabel {
-    padding-inline-start: 3px;
+.loadingInput {
+  &::after {
+    content: "";
+    position: absolute;
+    width: 16px;
+    height: 16px;
+    top: calc(50% - 8px);
+    background-color: var(--toolbar-icon-bg-color);
+    mask-size: cover;
+    mask-image: var(--loading-icon);
   }
 
-  &::before {
-    inset-inline-start: -54px;
+  &.start::after {
+    inset-inline-start: 2px;
+  }
 
-    transition-property: visibility;
-    transition-delay: var(--loading-icon-delay);
+  &.end::after {
+    inset-inline-end: 2px;
   }
-  #pageNumber.loading + &::before {
-    visibility: visible;
+}
+
+#numPages {
+  &.toolbarLabel {
+    padding-inline-start: 3px;
   }
 }
 
diff --git a/web/viewer.html b/web/viewer.html
index 46a767396..0eee77267 100644
--- a/web/viewer.html
+++ b/web/viewer.html
@@ -138,7 +138,9 @@ See https://github.com/adobe-type-tools/cmap-resources
       <div id="mainContainer">
         <div class="findbar hidden doorHanger" id="findbar">
           <div id="findbarInputContainer">
-            <input id="findInput" class="toolbarField" title="Find" placeholder="Find in document…" tabindex="91" data-l10n-id="pdfjs-find-input" aria-invalid="false">
+            <span class="loadingInput end">
+              <input id="findInput" class="toolbarField" title="Find" placeholder="Find in document…" tabindex="91" data-l10n-id="pdfjs-find-input" aria-invalid="false">
+            </span>
             <div id="findPrevNext" class="splitToolbarButton">
               <button id="findPrevious" class="toolbarButton" title="Find the previous occurrence of the phrase" tabindex="92" data-l10n-id="pdfjs-find-previous-button">
                 <span data-l10n-id="pdfjs-find-previous-button-label">Previous</span>
@@ -325,7 +327,9 @@ See https://github.com/adobe-type-tools/cmap-resources
                     <span data-l10n-id="pdfjs-next-button-label">Next</span>
                   </button>
                 </div>
-                <input type="number" id="pageNumber" class="toolbarField" title="Page" value="1" min="1" tabindex="15" data-l10n-id="pdfjs-page-input" autocomplete="off">
+                <span class="loadingInput start">
+                  <input type="number" id="pageNumber" class="toolbarField" title="Page" value="1" min="1" tabindex="15" data-l10n-id="pdfjs-page-input" autocomplete="off">
+                </span>
                 <span id="numPages" class="toolbarLabel"></span>
               </div>
               <div id="toolbarViewerRight">
```
That said, your refactoring is good enough by itself to accept this patch as it is, but I really think that we should remove this kind of "strange" numbers.

Reply:
That's clever, thank you!